### PR TITLE
Update dependencies, fix bug, use new trigger data

### LINKED ISSUE #18: Workflow fails to create a GitHub issue
When I deploy this app to Slack infra, and kick off the link trigger, fill out the form, no issue is created. In `slack activity` logs I see:

```
Token fetch succeeded for user 'U02AEHE4KG9' on workspace 'T029V6468RL' for app 'A052K0SBGBV' and provider 'github'
2023-04-10 16:27:38 [info] [Wf052MJEG6QL] (Trace=Tr052XUJK82D) Execution started for workflow 'Create new issue'
2023-04-10 16:27:38 [info] [Wf052MJEG6QL] (Trace=Tr052XUJK82D) Executing workflow step 1 of 3
2023-04-10 16:27:38 [info] [Fn010N] (Trace=Tr052XUJK82D) Function execution started for builtin function 'Open a form'
2023-04-10 16:27:47 [info] [Fn010N] (Trace=Tr052XUJK82D) Function execution completed for function 'Open a form'
2023-04-10 16:27:48 [info] [Wf052MJEG6QL] (Trace=Tr052XUJK82D) Execution completed for workflow step 'Open a form'
Token fetch succeeded for user 'U02AEHE4KG9' on workspace 'T029V6468RL' for app 'A052K0SBGBV' and provider 'github'
2023-04-10 16:27:49 [error] [Wf052MJEG6QL] (Trace=Tr052XUJK82D) Execution failed for workflow 'Create new issue'
	parameter_validation_failed - Validation for parameter `githubIssue` failed: missing_required_value
Token fetch succeeded for user 'U02AEHE4KG9' on workspace 'T029V6468RL' for app 'A052K0SBGBV' and provider 'github'
2023-04-10 16:28:50 [error] [Wf052MJEG6QL] (Trace=Tr052XUJK82D) Execution failed for workflow 'Create new issue'
	parameter_validation_failed - Validation for parameter `githubIssue` failed: missing_required_value
Token fetch succeeded for user 'U02AEHE4KG9' on workspace 'T029V6468RL' for app 'A052K0SBGBV' and provider 'github'
2023-04-10 16:29:12 [error] [Wf052MJEG6QL] (Trace=Tr052XU78T4H) Execution failed for workflow 'Create new issue'
	parameter_validation_failed - Validation for parameter `githubIssue` failed: missing_required_value
Token fetch succeeded for user 'U02AEHE4KG9' on workspace 'T029V6468RL' for app 'A052K0SBGBV' and provider 'github'
2023-04-10 16:30:50 [error] [Wf052MJEG6QL] (Trace=Tr052XUJK82D) Execution failed for workflow 'Create new issue'
	parameter_validation_failed - Validation for parameter `githubIssue` failed: missing_required_value
Token fetch succeeded for user 'U02AEHE4KG9' on workspace 'T029V6468RL' for app 'A052K0SBGBV' and provider 'github'
2023-04-10 16:32:13 [error] [Wf052MJEG6QL] (Trace=Tr052XU78T4H) Execution failed for workflow 'Create new issue'
	parameter_validation_failed - Validation for parameter `githubIssue` failed: missing_required_value
```

This same behaviour is reproducible in local apps, too.

### PR DESCRIPTION
**Pull request by filmaj**
- Updates SDK and API dependencies to latest versions
- Make explicit in the README to create a GitHub OAuth app on GitHub - different from a GitHub App
- Fixes #18 by making all GitHub issue fields required - works around #18
- Uses the latest SDK version's enumerated Trigger properties instead of curly-brace string incantations
- Removes deno.lock file

### CODE REVIEW

File: functions/create_issue.ts [hunk 1]
```diff
@@ -46,7 +46,7 @@ export const CreateIssueDefinition = DefineFunction({
             description: "Assignees",
           },
         },
-        required: ["title"],
+        required: ["title", "description", "assignees"],
```
Comment (e-zim) [inline functions/create_issue.ts:49]: Nice fix! I am still hoping that `DefineProperty` will allow non-required, `nullable` fields, but this seems like a perfect fix for the now!

File: triggers/create_new_issue_shortcut.ts [hunk 2]
```diff
@@ -9,16 +10,16 @@ import CreateNewIssueWorkflow from "../workflows/create_new_issue.ts";
...
   description: "Create a new GitHub issue in a repository",
   workflow: `#/workflows/${CreateNewIssueWorkflow.definition.callback_id}`,
   inputs: {
     interactivity: {
-      value: "{{data.interactivity}}",
+      value: TriggerContextData.Shortcut.interactivity,
     },
     channel: {
-      value: "{{data.channel_id}}",
+      value: TriggerContextData.Shortcut.channel_id,
```
Comment (e-zim) [inline triggers/create_new_issue_shortcut.ts:22]: I'm pretty dang happy to see this improvement. So much safety with these types 🙌

**Review by e-zim — APPROVED** [review]
Thank you for these improvements! I left the version updates to #24 (so we might want to update the PR title) but everything here looks and works wonderfully :shipit:

File: functions/create_issue.ts (hunk 1 above)
Comment (filmaj) [reply functions/create_issue.ts:49]: Yeah unfortunately the backend discussion around optional object properties and the nullability enforcement died with the backend team

Comment (e-zim) [reply functions/create_issue.ts:49]: Ahh unfortunate for sure... Should we consider #18 closed with these changes then?

**Pull request merged**